src/main/java/edu/umich/tl/analytics/CanvasCourseReportFilter.java

```diff
@@ -0,0 +1,197 @@
+package edu.umich.tl.analytics;
+
+import java.io.IOException;
+import java.util.Hashtable;
+import java.util.Properties;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import edu.umich.tl.utils.ReportUtilities;
+
+
+public class CanvasCourseReportFilter implements Filter {
+	private static Log M_log = LogFactory.getLog(CanvasCourseReportFilter.class);
+	private static final String OU_GROUPS = "ou=Groups";
+	private static final String LDAP_CTX_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+	private static final String SYSTEM_PROPERTY_FILE_PATH = "canvasCourseReportPath";
+	protected static Properties appExtPropertiesFile;
+	protected static final String PROPERTY_CANVAS_URL = "canvas.url";
+	protected static final String PROPERTY_CANVAS_ADMIN="canvas.admin.token";
+	protected static final String PROPERTY_USE_TEST_URL = "use.test.url";
+	protected static final String PROPERTY_LDAP_SERVER_URL = "ldap.server.url";
+	private static final String PROPERTY_AUTH_GROUP = "mcomm.group";
+	private String providerURL = null;
+	private String mcommunityGroup = null;
+	protected static String canvasURL = null;
+	protected static String canvasToken = null;
+	private boolean isTestUrlEnabled=false;
+	private static final String FALSE = "false";
+	private static final String TEST_USER = "testUser";
+
+	public void init(FilterConfig arg0) throws ServletException {
+		M_log.debug("Filter init(): called");
+		getExternalAppProperties();
+	}
+
+	private void getExternalAppProperties() {
+		M_log.debug("getExternalAppProperties(): called");
+		String propertiesFilePath = System.getProperty(SYSTEM_PROPERTY_FILE_PATH);
+		if (!ReportUtilities.isEmpty(propertiesFilePath)) {
+			appExtPropertiesFile=ReportUtilities.getPropertiesFromURL(propertiesFilePath);
+			if(appExtPropertiesFile!=null) {
```

Comment:
For clarity just test to see if the value is null and, if so, log a message and return.  This avoids a lot of un-informative nesting.